transformer_engine/pytorch/ops/basic/all_gather.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+"""Fusable operation for all-gather."""
+
+from __future__ import annotations
+from typing import Optional
+
+import torch
+
+from transformer_engine.pytorch.float8_tensor import Float8Tensor
+from transformer_engine.pytorch.ops.op import (
+    BasicOperation,
+    OperationContext,
+)
+from .._common import convert_tensor, is_float8_tensor
+
+
+class AllGather(BasicOperation):
+    """All-gather tensor along outer dimension
+
+    Equivalent to gathering tensors from all processes and
+    concatenating along the first dimension.
+
+    Parameters
+    ----------
+    process_group: torch.distributed.ProcessGroup, default = world group
+        Process group for communication
+
+    """
+
+    def __init__(
+        self,
+        process_group: Optional[torch.distributed.ProcessGroup] = None,
+    ) -> None:
+        super().__init__()
+        self.process_group: Optional[torch.distributed.ProcessGroup] = process_group
+        self.process_group_size: int = torch.distributed.get_world_size(process_group)
+
+    def op_forward(
+        self,
+        ctx: OperationContext,
+        input: torch.Tensor,  # pylint: disable=redefined-builtin
```

Comment:
These are worth changing IMO.
`input` &rarr; `inp`

Reply:
I'd agree that's true for internal implementations, but `input` feels much better for a user-facing API:
```
op = te.ops.AllGather()
y = op(input=x)
```